agent/utils/retry/retry.go

```diff
@@ -132,7 +133,7 @@ func WaitForDurationWithContext(eventFlowCtx context.Context, delay time.Duratio
 		})
 		return true
 
-	case <-eventFlowCtx.Done():
+	case <-(*eventFlowCtx).Done():
```

Comment:
I think you can still use `eventFlowCtx.Done()` here

Reply:
If I use eventFlowCtx.Done() I get an error saying .Done() is undefined- i guess that's because eventFlowCtx is a pointer

Reply:
I thought golang automatically dereference the pointer when it's selector expression - https://go.dev/ref/spec#Selectors, but nvm then